src/api/authorityApi.ts

```diff
@@ -0,0 +1,80 @@
+import Axios from 'axios';
+import { Dispatch } from 'redux';
+
+import { addNotification } from '../redux/actions/notificationActions';
+import i18n from '../translations/i18n';
+import { Authority } from '../types/authority.types';
+import { StatusCode, USE_MOCK_DATA } from '../utils/constants';
+import { getIdToken } from './userApi';
+
+export const getAuthorities = async (name: string, dispatch: Dispatch) => {
+  const url = '/authority';
+
+  // remove when Authorization headers are set for all requests
+  const idToken = USE_MOCK_DATA ? '' : await getIdToken();
+  const headers = {
+    Authorization: `Bearer ${idToken}`,
+  };
+
+  try {
+    const response = await Axios.post(url, { name }, { headers });
+
+    if (response.status === StatusCode.OK) {
+      return response.data;
+    } else {
+      dispatch(addNotification(i18n.t('feedback:error.get_authorities'), 'error'));
+    }
+  } catch {
+    dispatch(addNotification(i18n.t('feedback:error.get_authorities'), 'error'));
+  }
+};
+
+export const getAuthorityByFeideId = async (feideId: string, dispatch: Dispatch) => {
+  const url = '/authority';
+
+  // remove when Authorization headers are set for all requests
+  const idToken = USE_MOCK_DATA ? '' : await getIdToken();
+  const headers = {
+    Authorization: `Bearer ${idToken}`,
+  };
+
+  try {
+    const response = await Axios.post(url, { feideId }, { headers });
+
+    if (response.status === StatusCode.OK) {
+      const filteredAuthorities = response.data.filter((auth: Authority) => auth.feideId === feideId);
+      return filteredAuthorities?.[0] ?? null;
+    } else {
+      dispatch(addNotification(i18n.t('feedback:error.get_authority'), 'error'));
+    }
+  } catch {
+    dispatch(addNotification(i18n.t('feedback:error.get_authority'), 'error'));
+  }
+};
+
+export const updateAuthority = async (authority: Partial<Authority> | null, dispatch: Dispatch) => {
+  if (!authority) {
```

Comment:
mer et TS-spørsmål: er det ikke like greit å bare ta inn en Authority uten Partial og null, så vil TS klage hvis vi prøver å kalle denne med ugyldig parameter?

Reply:
feideId, orcId, birthDate er optional hvertfall. mulig flere felt også er det (handle)? men jeg er enig i at vi kan heller spesifisere i authority-typen at de er optional slik at TS ikke tolker alle som optional

Reply:
når jeg tenker meg om: hvis vi bestemmer oss for å sette typen her ikke-partial, så kan det bli vanskelig å flytte orcId til user.authority.orcId uten at det finnes name og scn. tror dessverre Authority må være Partial. eventuelt kan vi lage to forskjellige typer. slik at svaret fra /authority er 
`name, scn, handle <- påkrevd`
`orcId, feideId, birthDate <- optional`, mens selve Authority-typen er partial
@elkebab ?